internal/chain/engine.go

```diff
@@ -215,9 +216,25 @@ func (e *engineImpl) VerifySeal(chain engine.ChainReader, header *block.Header)
 		return errors.Wrapf(err,
 			"cannot calculate quorum for block %s", header.Number())
 	}
-	if count := utils.CountOneBits(mask.Bitmap); count < int64(parentQuorum) {
-		return ctxerror.New("[VerifySeal] Not enough signature in LastCommitSignature from Block Header",
-			"need", parentQuorum, "got", count)
+	if chain.Config().IsStaking(parentHeader.Epoch()) {
+		slotList, err := shard.DecodeWrapper(parentHeader.ShardState())
```

Comment:
can directly use header.GetShardState()

Reply:
Not really. This should be "slotList, err := chain.ReadShardState(parentHeader.Epoch())"